### PR TITLE
test: re-enable CLI test support by using EncodeDecimal in json.dumps()

### PR DESCRIPTION
**Pull request by fanquake**
As mentioned in https://github.com/bitcoin/bitcoin/pull/17675#issuecomment-563188648.

### CODE REVIEW

**Comment by practicalswift** [comment]
Thanks for following up on this!

ACK b6f9e3576a1ea18572e4803aeb3f39330f0cb759 assuming Travis is happy too -- diff looks correct :)

File: test/functional/test_framework/util.py [hunk 1]
```diff
@@ -190,6 +190,11 @@ def check_json_precision():
     if satoshis != 2000000000000003:
         raise RuntimeError("JSON encode/decode loses precision")
 
+def EncodeDecimal(o):
+    if isinstance(o, Decimal):
+        return str(o)
+    raise TypeError(repr(o) + " is not JSON serializable")
```
Comment (MarcoFalke) [inline test/functional/test_framework/util.py]: Any reason why this is duplicated from the identical function in authproxy?

Comment (fanquake) [reply test/functional/test_framework/util.py]: There were include issues running tests directly when I moved the authproxy usage into util. I can take another look.

Comment (MarcoFalke) [reply test/functional/test_framework/util.py]: ok, you may leave it as is

**Review by maflcko — APPROVED** [review]
> ACK b6f9e35 assuming Travis is happy too -- diff looks correct :)

travis doesn't run with `--usecli`

**Comment by practicalswift** [comment]
@MarcoFalke I think it does for `wallet_multiwallet.py`, `wallet_createwallet.py`, `wallet_watchonly.py` and `rpc_deriveaddresses.py`? :)

**Comment by MarcoFalke** [comment]
> @MarcoFalke I think it does for wallet_multiwallet.py, wallet_createwallet.py, wallet_watchonly.py and rpc_deriveaddresses.py? :)

Yeah, but none of these files are touched in this pull

**Comment by practicalswift** [comment]
@MarcoFalke They all use `arg_to_cli` which is touched by this pull, no? :)

**Pull request merged**